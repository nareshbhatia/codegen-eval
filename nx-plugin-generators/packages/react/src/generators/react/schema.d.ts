export interface ReactGeneratorSchema {
  name: string;
  tags?: string;
  directory?: string;
}
