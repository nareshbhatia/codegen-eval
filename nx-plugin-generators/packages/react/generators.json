{
  "$schema": "http://json-schema.org/schema",
  "name": "react",
  "version": "0.0.1",
  "generators": {
    "react": {
      "factory": "./src/generators/react/generator",
      "schema": "./src/generators/react/schema.json",
      "description": "react generator"
    },
    "component": {
      "factory": "./src/generators/component/generator",
      "schema": "./src/generators/component/schema.json",
      "description": "component generator"
    },
    "context": {
      "factory": "./src/generators/context/generator",
      "schema": "./src/generators/context/schema.json",
      "description": "context generator"
    }
  }
}
