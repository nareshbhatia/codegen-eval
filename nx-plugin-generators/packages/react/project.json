{
  "root": "packages/react",
  "sourceRoot": "packages/react/src",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/react",
        "main": "packages/react/src/index.ts",
        "tsConfig": "packages/react/tsconfig.lib.json",
        "assets": [
          "packages/react/*.md",
          {
            "input": "./packages/react/src",
            "glob": "**/!(*.ts)",
            "output": "./src"
          },
          {
            "input": "./packages/react/src",
            "glob": "**/*.d.ts",
            "output": "./src"
          },
          {
            "input": "./packages/react",
            "glob": "generators.json",
            "output": "."
          },
          {
            "input": "./packages/react",
            "glob": "executors.json",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/react/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/packages/react"],
      "options": {
        "jestConfig": "packages/react/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
