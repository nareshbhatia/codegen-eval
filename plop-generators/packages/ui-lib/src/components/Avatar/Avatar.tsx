interface AvatarProps {
  name: string;
  photo: string;
}

export function Avatar({ name, photo }: AvatarProps) {
  return (
    <img
      className="inline-block h-56 w-56 rounded-full"
      src={photo}
      alt={name}
    />
  );
}
