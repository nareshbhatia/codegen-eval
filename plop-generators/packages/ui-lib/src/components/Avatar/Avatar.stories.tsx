import * as React from 'react';
import { Story, Meta } from '@storybook/react';
import { Avatar } from './Avatar';

export default {
  title: 'Components/Avatar',
  component: Avatar,
} as Meta;

const Template: Story = (args) => (
  <Avatar name={args.name} photo={args.photo} />
);

export const AvatarStory = Template.bind({});
AvatarStory.storyName = 'Avatar';
AvatarStory.args = {
  name: 'Paul Silva',
  photo:
    'https://images.unsplash.com/photo-1568585105565-e372998a195d?w=256&h=256&fit=crop&crop=entropy',
};
