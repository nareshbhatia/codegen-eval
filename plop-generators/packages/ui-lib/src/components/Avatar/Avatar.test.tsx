import { render } from '@testing-library/react';
import { Avatar } from './Avatar';

describe('<Avatar />', () => {
  test('renders correctly', () => {
    const { asFragment } = render(
      <Avatar
        name="Paul Silva"
        photo="https://images.unsplash.com/photo-1568585105565-e372998a195d?w=256&h=256&fit=crop&crop=entropy"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
