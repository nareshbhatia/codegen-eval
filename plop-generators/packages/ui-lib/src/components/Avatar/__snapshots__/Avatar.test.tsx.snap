// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> renders correctly 1`] = `
<DocumentFragment>
  <img
    alt="Paul Silva"
    class="inline-block h-56 w-56 rounded-full"
    src="https://images.unsplash.com/photo-1568585105565-e372998a195d?w=256&h=256&fit=crop&crop=entropy"
  />
</DocumentFragment>
`;
